### ISSUE TITLE
Problems trying to use 3 controllers

### ISSUE THREAD

**Issue by MetalMilitia623** [x360ce/x360ce#68]
I am trying to use 3 controllers to play speedrunners, jamestown, serious sam 3, etc. on local multi player. I have A 360 controller, A ps3 controller and an original xbox controller s.

They were all working in x360ce but when loading speedrunners only the first control to enter an input would work. However now after my pc has gone to sleep and been woken up the ps3 controller either appears as controller 2 and 3 or the 360 controller is in number 2 but the left joystick gets stuck in all directions but there is no problem with the actual joystick or while using it directly in any other game.

I'm not sure if the issue is something with the SPC drivers for ps3 controllers and x360ce but it is very strange and any ideas would be appreciated.

**Comment by Squall-Leonhart** [comment]
what dll version?

can you post a log?

**Comment by MetalMilitia623** [comment]
I have .dll version 3.3.1.444 and I enabled logs but am not sure where it is putting them.

**Comment by Squall-Leonhart** [comment]
Application is at v3.0.2.25 Beta 2 (2015-04-30) and Libraries are at 3.4.1.1245 (2015-04-30)

Please open a new ticke if the issue still occurs when you aren't using 3-4 year old libaries.

**Comment by MetalMilitia623** [comment]
Issue still persists with the new stuff.. 

May have solved it by turning pass through on on the controller 2 that was sticking for no reason. That seems to have stopped it from sticking and it still works despite pass through being supposed to turn it off.

They all now work properly in the x360ce software but no in speedrunners, only the first controller to input will work. going to try some other games now.

**Comment by Squall-Leonhart** [comment]
passthrough forwards the controllers native xinput capabilities to the game.

open up the joypad control panel (joy.cpl) and set the device that appears as the first controller as the preferred controller.

Speedrunners appears to be Unity 3D, games on this toolkit/engine do not use xinput natively unless otherwise using a custom xinput manager.

Please contact the developer and direct him towards http://forum.unity3d.com/threads/incontrol-input-and-cross-platform-controller-support-for-unity-made-easy.257619/

**Comment by Squall-Leonhart** [comment]
Or maybe it isn't....

hmm. i can't find much information about the games engine.

**Comment by MetalMilitia623** [comment]
Yea, I had the same issue the other day, was asking a bunch of people what the game is made on. 

The reason this issue is so strange is all three controllers were working for a short period of time but a tab out of the game caused one of them to stop and then trying to get it back caused more problems and now i am here where only one will work in that game.

**Comment by Squall-Leonhart** [comment]
It appears to be using XNA on windows, 

Try configuring inputhook as follows

[InputHook]
HookLL=0
HookDI=1
HookSA=0
HookWT=0
HOOKPIDVID=1
HookNAME=1
HookSTOP=0
//Timeout=120

**Comment by MetalMilitia623** [comment]
Did not work, would the x360ce log for speedrunners help? and if so what's the best way to get in on here, just copy paste?

Wait, in the program with that in the ini it says hook mode disabled, is that correct?

**Comment by Squall-Leonhart** [comment]
hookmode is only relevant to ancient versions of the dll, Ejocy should really remove it from the app.

Pasting the log in pastebin would be most welcome, if it is overly large.

**Comment by Squall-Leonhart** [comment]
Am downloading jamestown presently to take a look.

**Comment by Squall-Leonhart** [comment]
well im getting two controllers to work ok, but if i press one of the triggers the stick stops being able to move the player - only dpad works unless i move stick in a direction and then press either trigger, then the player gets stuck moving in the direction of the stick.

**Comment by Squall-Leonhart** [comment]
jamestown is not using xinput at all,

it references xinput1_3, it lets us create a log but uses dinput only, x360ce is not even given a chance to project an xinput compatible gamepad.

**Comment by Squall-Leonhart** [comment]
no trouble with splitscreen on serious sam with x86 xinput1_3.dll

**Comment by MetalMilitia623** [comment]
Strange, Jamestown and serious sam work perfectly for me. I'll have to find some more local mp games to try but as of now it seems as speedrunners is the only game that doesn't work. If you want to troubleshoot there is a demo that is the offline portion of the game only. 

Risk of rain is also working perfectly for me. 

So far it's the two early acess games that are giving me issues. Speedrunners (SR) which will only see the first controller to input and road redemption (RR) which just gets stuck always going up in the menu and you are forced to just close the game. I don't really care about making x360ce work with RR at the moment just really want to be able to play speedrunners locally because the netcode still needs work and it is way more fun local.

Still it is very strange that you are having an issue with Jamestown where i am not.

**Comment by Squall-Leonhart** [comment]
Jamestown simply doesn't need x360ce at all, it doesn't use it.

Road Redemption is definitely Unity 3D, But they are using a custom xinput manager going by the discussions,  what you are seeing sounds like typical RawInput behavior though...

But this combo should work

[InputHook]
HookDI=1
HOOKPIDVID=1
HookNAME=1

**Comment by MetalMilitia623** [comment]
Jamestown does need it for me because it doesn't see my original Xbox Controller S without it. Also it helps with the deadzones so I can zero the sticks. 

Those lines of code did not work for Road Redemption. 

Man this is pretty frustrating. I've never had a problem with this program if everything worked in the ui it always worked in game.

**Comment by Nucleoprotein** [comment]
I'm almost sure that Jamestown is using Xinput, at least Steam version of Jamestown.

**Comment by MetalMilitia623** [comment]
Good news, speedrunners now works. Idk what happened to make it work but it works now as it should.

**Comment by Squall-Leonhart** [comment]
"I'm almost sure that Jamestown is using Xinput, at least Steam version of Jamestown."

Steam version has Xinput1_3 in its dll references, it loads our dll, creates a log with only the GDB location in it, but no hook sound, and uses Dinput instead.

both my MLG 360 pad and my S type works fine in jamestown without x360ce, just has the trigger issue,

**Comment by MetalMilitia623** [comment]
The road redemption forums on steam and their personal ones have been of no help. Still trying to figure out why when using this software that game is always seeing the stick going up. Every other game seems to work with no issues now. The updates files did help quite a bit, lol.

**Comment by Nucleoprotein** [comment]
Set bigger deadzone ?

**Comment by MetalMilitia623** [comment]
Nope, in the x360 ui all controllers are showing as being centered perfectly.

**Comment by Squall-Leonhart** [comment]
reason it see's the stick going up is because it ignores Xinput and maps the raw information from the driver, it incorrectly specifies a dinput analog trigger as pressed and keeps scrolling up.

**Comment by MetalMilitia623** [comment]
That may be true but it was something in the game, since they switched to unity 5 it works perfectly with x360ce.

**Comment by Squall-Leonhart** [comment]
looks like they actually fixed some 4 year old bugs.

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
